packages/realm/src/Object.ts

```diff
@@ -236,25 +228,39 @@ export class RealmObject<T = DefaultObject> {
   }
 
   /**
+   * @internal
    * The Realm managing the object.
+   * Note: this is on the injected prototype from ClassMap.defineProperties().
    */
-  public readonly realm!: Realm;
+  public declare readonly [REALM]: Realm;
 
   /**
    * @internal
    * The object's representation in the binding.
    */
-  public readonly [INTERNAL]!: binding.Obj;
+  public declare readonly [INTERNAL]: binding.Obj;
+
+  /**
+   * @internal
+   * Lazily created wrapper for the object notifier.
+   */
+  private declare [INTERNAL_LISTENERS]: ObjectListeners<T> | null;
+
+  /**
+   * @internal
+   * Note: this is on the injected prototype from ClassMap.defineProperties()
+   */
+  private declare readonly [KEY_ARRAY]: ReadonlyArray<string>;
 
   /**
    * @internal
-   * Wrapper for the object notifier.
+   * Note: this is on the injected prototype from ClassMap.defineProperties()
    */
-  private readonly [INTERNAL_LISTENERS]!: ObjectListeners<T>;
+  private declare readonly [KEY_SET]: ReadonlySet<string>;
 
-  // TODO: Find a way to bind this in
   keys(): string[] {
-    throw new Error("This is expected to have a per-class implementation");
+    // copying to prevent caller from modifying the static array.
+    return [...this[KEY_ARRAY]];
   }
```

Comment:
I wonder if `Array.from` is any faster as it potentially won't have to iterate the array data.

Reply:
[good idea](https://perf.link/#eyJpZCI6IjRyNjhyYmdkbzlhIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoMTAwMCkua2V5cygpXSIsInRlc3RzIjpbeyJuYW1lIjoic3ByZWFkIiwiY29kZSI6IlsuLi5kYXRhXSIsInJ1bnMiOlszNzAwMCw5MTAwMCwyMjUwMDAsMjY3MDAwLDIyNTAwMCwyMjUwMDAsMTQwMDAwLDUzMDAwLDI4MzAwMCwyMjUwMDAsMzMyMDAwLDIyNTAwMCw0MDgwMDAsMjkwMDAwLDIyNTAwMCwxNzkwMDAsMjYwMDAsMjIzMDAwLDk4MDAwLDIyNTAwMCwzMzkwMDAsMjI1MDAwLDE4NTAwMCw0NTAwMCw3MTAwMCw2ODAwMCwzMzQwMDAsMjAwMCwyMDcwMDAsMjI1MDAwLDIzMDAwLDIyNTAwMCwzNTEwMDAsNTcwMDAsMjI1MDAwLDQwMjAwMCwyMjUwMDAsMjkwMDAsODMwMDAsMTQyMDAwLDM3MDAwLDIyNTAwMCwyMjUwMDAsNDU2MDAwLDEyOTAwMCwzNjEwMDAsNDU2MDAwLDQ1NjAwMCwyMjUwMDAsNDQzMDAwLDM2NDAwMCwyMjUwMDAsMjY2MDAwLDIzNzAwMCw0NTcwMDAsMjcxMDAwLDEyODAwMCwyMjUwMDAsMjI1MDAwLDIxNjAwMCwyMjUwMDAsMTAyMDAwLDE1MDAwLDk5MDAwLDQ2MDAwMCwyMDUwMDAsNzEwMDAsNDgwMDAsMjExMDAwLDMwNzAwMCw0NTYwMDAsMTM3MDAwLDExNDAwMCwzMzQwMDAsNDQ3MDAwLDYwMjAwMCw0NTYwMDAsMTI2MDAwLDI4MDAwLDMyNzAwMCwxNDUwMDAsMjQ3MDAwLDM1NDAwMCwxOTcwMDAsMTM4MDAwLDQ1NzAwMCwzMDEwMDAsMzkwMDAsMjI1MDAwLDIyNTAwMCwyMjUwMDAsMjI1MDAwLDM5OTAwMCwyMDAwMCwzNjYwMDAsMjg4MDAwLDkwMDAsNDU2MDAwLDk1MDAwLDc3MDAwXSwib3BzIjoyMjM1MDB9LHsibmFtZSI6ImFycmF5LmZyb20iLCJjb2RlIjoiQXJyYXkuZnJvbShkYXRhKSIsInJ1bnMiOlsyMjUwMDAsMTQ2MDAwLDIyNTAwMCwyMjUwMDAsMTAwMCw0MzIwMDAsMzY5MDAwLDI5ODAwMCwzNTEwMDAsMTkyMDAwLDgwMDAsMTEzMDAwLDQyMjAwMCwyMjUwMDAsMjUyMDAwLDI4MDAwMCw0NDMwMDAsOTAwMCwyNTEwMDAsMjI1MDAwLDIyNTAwMCwyMDAwMCwyMjUwMDAsMjI1MDAwLDIyNjAwMCwyMjUwMDAsNTYwMDAwLDM4NTAwMCwyNDUwMDAsNDU2MDAwLDM5MDAwLDIyNTAwMCw0MzIwMDAsNzMwMDAsMTkzMDAwLDI4ODAwMCwzMTIwMDAsMjEzMDAwLDQ2MjAwMCw0NzAwMCwyOTEwMDAsMjI1MDAwLDIyNTAwMCwxNzAwMCwyMjUwMDAsMzQyMDAwLDIyNTAwMCwzODUwMDAsMjI1MDAwLDE4NzAwMCwzOTQwMDAsMjI1MDAwLDQyNzAwMCwyMjUwMDAsMjA3MDAwLDEzNjAwMCw0NTYwMDAsMTEzMDAwLDMwMjAwMCwyMTAwMCwyNzkwMDAsMzI5MDAwLDM3NjAwMCw0NTYwMDAsMjAzMDAwLDMwMDAwLDQ4MDAwMCwzNDIwMDAsNDMxMDAwLDEyNjAwMCwyNjcwMDAsNjYyMDAwLDIwMzAwMCwyMjUwMDAsMzkxMDAwLDI5MzAwMCwyMjUwMDAsNTAwMCw0NjQwMDAsMTUzMDAwLDIxNjAwMCwzNzIwMDAsMjIzMDAwLDMzMzAwMCwzODEwMDAsMjI1MDAwLDI1MjAwMCwxMzMwMDAsMjI1MDAwLDMzMDAwLDQ1NjAwMCw0MzMwMDAsNjAwMDAsMzI5MDAwLDIyMTAwMCw0NTYwMDAsMzc2MDAwLDMzMDAwLDM0MjAwMCw1MzAwMF0sIm9wcyI6MjU3MTMwfV0sInVwZGF0ZWQiOiIyMDIyLTEyLTE5VDE1OjQyOjExLjI3OVoifQ%3D%3D)

Actually, even thought it is slightly faster in my browser (edge, which I think uses v8), `Array.from` seems to be a bit slower in both node 16 and 18. So I think I'll stick with `[...arr]` for now. It also has the benefit of being obvious what it does, while at least I needed to look up the behavior of `Array.from`